Description: Choose character device for "-c" testing more safely
 Hopefully solves test suite failures on Hurd
Author: Pino Toscano <pino@debian.org>
Reviewed-By: Axel Beckert <abe@debian.org>

Index: zsh/Test/C02cond.ztst
===================================================================
--- zsh.orig/Test/C02cond.ztst	2013-09-20 16:21:32.000000000 +0200
+++ zsh/Test/C02cond.ztst	2013-09-20 17:59:03.000000000 +0200
@@ -41,9 +41,14 @@
   fi
 0D:-b cond
 
-  # Use hardcoded /dev/tty because globbing inside /dev fails on Cygwin
-  char=/dev/tty
-  [[ -c $char && ! -c $zerolength ]]
+  # Hardcoded /dev/tty as necessary for Cygwin may fail on Hurd
+  tty=$(find /dev/ -nowarn -maxdepth 1 -name 'tty*' -type c -print)
+  if [[ -n $tty ]]; then
+    [[ -c $tty[(f)1] && ! -c $zerolength ]]
+  else
+    print -u$ZTST_fd 'Warning: Not testing [[ -c tty ]] (no tty found)'
+    [[ ! -c $zerolength ]]
+  fi
 0:-c cond
 
   [[ -d . && ! -d zerolength ]]
